localstack-core/localstack/services/kms/provider.py

```diff
@@ -1354,6 +1357,74 @@ def enable_key_rotation(
+            context.region,
+            request.get("KeyId"),
+            disabled_key_allowed=True,
+            pending_deletion_key_allowed=True,
+        )
+
+        include_key_material = request.get("IncludeKeyMaterial")
+        if include_key_material is not None:
+            if include_key_material not in ["ALL_KEY_MATERIAL", "ROTATIONS_ONLY"]:
+                raise ValidationException(
+                    f"1 validation error detected: Value '{include_key_material}' at 'includeKeyMaterial' "
+                    f"failed to satisfy constraint: Member must satisfy enum value set: [ALL_KEY_MATERIAL, ROTATIONS_ONLY]"
+                )
+
+            if key.metadata["KeySpec"] != KeySpec.SYMMETRIC_DEFAULT:
+                raise ValidationException(
+                    "Input parameter includeKeyMaterial can only be used with symmetric keys."
+                )
+
+            self._validate_key_supports_include_key_material(key)
+
+        rotation_history = []
+        if key.supports_rotation() or include_key_material is not None:
+            for rotation in key.key_rotations:
+                rotation_entry = {"KeyId": rotation.key_id}
+
+                if rotation.key_material_id is not None:
+                    rotation_entry["KeyMaterialId"] = rotation.key_material_id
+                if rotation.key_material_state is not None:
+                    rotation_entry["KeyMaterialState"] = rotation.key_material_state
+                if rotation.key_material_description is not None:
+                    rotation_entry["KeyMaterialDescription"] = rotation.key_material_description
+                if rotation.rotation_date is not None:
+                    rotation_entry["RotationDate"] = rotation.rotation_date
+                if rotation.rotation_type is not None:
+                    rotation_entry["RotationType"] = rotation.rotation_type
+                if rotation.expiration_model is not None:
+                    rotation_entry["ExpirationModel"] = rotation.expiration_model
+                if rotation.import_state is not None:
+                    rotation_entry["ImportState"] = rotation.import_state
+                if rotation.valid_to is not None:
+                    rotation_entry["ValidTo"] = rotation.valid_to
+
+                if include_key_material == IncludeKeyMaterial.ALL_KEY_MATERIAL:
+                    rotation_history.append(rotation_entry)
+                else:  # Default ROTATIONS_ONLY
+                    if rotation.rotation_type in ["AUTOMATIC", "ON_DEMAND"]:
+                        rotation_history.append(rotation_entry)
+
+        rotations_list = PaginatedList(rotation_history)
+        page, next_token = rotations_list.get_page(
+            lambda rotation: rotation.get("RotationDate"),
+            next_token=request.get("Marker"),
+            page_size=request.get("Limit", 100),
+        )
+        kwargs = {"NextMarker": next_token, "Truncated": True} if next_token else {}
```

Comment:
I recommend here either we remove pagination for this PR and leave a TODO or add a test to validate this implementation.